File: Source Code/AdminOnlineFloralDelivery/src/app/service/event.service.ts
import { Injectable } from "@angular/core";
import { BaseURLService } from "./baseurl.service";
import { HttpClient } from "@angular/common/http";
import { lastValueFrom } from "rxjs";

@Injectable()
export class EventService {
    constructor(
        private BaseUrlService: BaseURLService,
        private HttpClient: HttpClient,
    ){}
    async findAll(){
        return await lastValueFrom(this.HttpClient.get(this.BaseUrlService.getBaseUrl()+'event/findallevent'));
    }
    async findId(id : string){
        return await lastValueFrom(this.HttpClient.get(this.BaseUrlService.getBaseUrl()+'event/SearchEventId/'+id));
    }
    async findStatus(status : string){
        return await lastValueFrom(this.HttpClient.get(this.BaseUrlService.getBaseUrl()+'event/findEventIsAction/'+status));
    }
    async findName(keyword : string){
        return await lastValueFrom(this.HttpClient.get(this.BaseUrlService.getBaseUrl()+'event/searchEventName/'+keyword));
    }
    async create(FormData: FormData){
        return await lastValueFrom(this.HttpClient.post(this.BaseUrlService.getBaseUrl()+'event/addEvent/',FormData));
    }
    async update(FormData: FormData){
        return await lastValueFrom(this.HttpClient.put(this.BaseUrlService.getBaseUrl()+'event/updateEvent',FormData));
    }

}

File: Source Code/DemoOnlineFloralDelivery/DemoOnlineFloralDelivery/Service/EventService.cs
using DemoOnlineFloralDelivery.Models;

namespace DemoOnlineFloralDelivery.Service;

public interface EventService
{
    //Hiển thị danh sách toàn bộ Event
    public dynamic findAllEvent();
    //Thêm(add) Event
    public bool addEvent(Event Event);
    //Cập nhật(update) Event
    public bool updateEvent(Event Event);
    //Hiển thị danh sách Event theo IsAction
    public dynamic findEventIsAction();
    //Tìm kiếm(search) Event theo Event_name
    public dynamic searcEventName(string keyword);
    //Tìm kiếm(search) Event theo Event_id
    public dynamic searchEventId(int id);
    public Event findEventId(int id);

    public dynamic findEventIsActionUser();

    public dynamic findEvenUserDealOfMonth();

}
